src/event-engine/presence/states/heartbeat_cooldown.ts

```diff
@@ -0,0 +1,54 @@
+import { State } from '../../core/state';
+import { Events, disconnect, joined, left, leftAll, timesUp } from '../events';
+import { Effects, leave, wait } from '../effects';
+import { HeartbeatingState } from './heartbeating';
+import { HeartbeatStoppedState } from './heartbeat_stopped';
+import { HeartbeatInactiveState } from './heartbeat_inactive';
+
+export type HeartbeatCooldownStateContext = {
+  channels: string[];
+  groups: string[];
+};
+
+export const HeartbeatCooldownState = new State<HeartbeatCooldownStateContext, Events, Effects>('HEARTBEATCOOLDOWN');
```

Comment:
Will it be better to use `HEARTBEAT_COOLDOWN`?

Reply:
Renamed ✔️ 
https://github.com/pubnub/javascript/blob/c35559fef8bfb49b63d334fd42fb3733f36c75bd/src/event-engine/presence/states/heartbeat_cooldown.ts#L13